packages/manager-react-components/src/hooks/datagrid/useIcebergV6.tsx

```diff
@@ -24,73 +19,48 @@ export function useResourcesIcebergV6<T = unknown>({
   queryKey,
   defaultSorting = undefined,
 }: IcebergFetchParamsV6 & IcebergV6Hook) {
-  const [totalCount, setTotalCount] = useState(0);
-  const [flattenData, setFlattenData] = useState<T[]>([]);
-  const [queryParams, setQueryParams] = useState<QueryParams>({
-    pageIndex: 1,
-    sorting: { desc: false, id: '' },
-  });
   const [sorting, setSorting] = useState<ColumnSort>(defaultSorting);
 
   const {
-    data,
+    data: dataSelected,
     fetchNextPage,
     isError,
     isLoading,
     error,
     status,
   } = useInfiniteQuery({
-    initialPageParam: null,
+    initialPageParam: 1,
     queryKey: [...queryKey, sorting],
-    queryFn: ({ pageParam }) =>
+    staleTime: Infinity,
+    retry: false,
+    queryFn: ({ pageParam: pageIndex }) =>
       fetchIcebergV6<T>({
         route,
         pageSize,
-        page: pageParam,
+        page: pageIndex,
         sortBy: sorting?.id || null,
         sortReverse: sorting?.desc,
       }),
-    staleTime: Infinity,
-    retry: false,
-    getNextPageParam: () => queryParams.pageIndex,
+    getNextPageParam: (lastPage, _allPages, lastPageIndex) => {
+      if (lastPage.totalCount / pageSize > lastPageIndex) return null;
+      return lastPageIndex + 1;
+    },
+    select: (data) => {
+      const pageIndex = data.pageParams?.[data.pageParams.length - 1] ?? 0;
+      const totalCount = data?.pages[0].totalCount ?? 0;
+      return {
+        data,
+        pageIndex,
+        totalCount,
+        flattenData: data?.pages.flatMap((page) => page.data) ?? [],
+        hasNextPage: totalCount / pageSize > pageIndex,
+      };
+    },
   });
 
-  useEffect(() => {
-    const flatten = data?.pages.map((page) => page.data).flat();
-    setFlattenData(flatten);
-  }, [data]);
-
-  useEffect(() => {
-    if (totalCount === 0 && data?.pages) {
-      setTotalCount(data.pages[0].totalCount);
-    }
-  }, [data]);
-
-  useEffect(() => {
-    if (sorting) {
-      setQueryParams(() => ({ pageIndex: 1, sorting }));
-    }
-  }, [sorting]);
-
-  useEffect(() => {
-    fetchNextPage();
-  }, [queryParams]);
-
-  const onFetchNextPage = () => {
-    setQueryParams((previousParams: QueryParams) => ({
-      ...previousParams,
-      pageIndex: previousParams.pageIndex + 1,
-    }));
-  };
-
   return {
-    data,
-    fetchNextPage: onFetchNextPage,
-    setFlattenData,
-    flattenData,
-    totalCount,
-    pageIndex: queryParams.pageIndex,
-    hasNextPage: totalCount / pageSize > queryParams.pageIndex,
+    ...dataSelected,
+    fetchNextPage,
```

Comment:
IMO, this is useless to return `isError, isLoading, isError ...` unless we want to constraint the properties returned by `useInfiniteQuery` to the returned object.
If not, I'll suggest to use rest parameter: 

```typescript
const {
    data,
   ...rest
  } = useInfiniteQuery(...)

return {
 pageIndex,
...,
...rest
}

```

Reply:
Yes you're right ! 👌